File: src/main/java/org/lovaprod/utils/MoneyUtils.java
package org.lovaprod.utils;

public class MoneyUtils {
    public static boolean isAmountValid(double amount){
        if (amount<0){
            System.out.println("Amount must be > 0"+MONEY_UNIT);
            return false;
        }
        return true;
    }

    public static boolean isAmountValid(double amount, double balance){
        if (!isAmountValid(amount)){
            return false;
        }
        if (amount > balance){
            System.out.println("amount must be < Balance: "+balance+MONEY_UNIT);
            return false;
        }
        return true;
    }

    public static boolean isAmountValid(double amount, double balance, double maxTransaction){
        if (!isAmountValid(amount,balance)){
            return false;
        }
        if (amount > maxTransaction){
            System.out.println("amount must be < maxTransaction: "+maxTransaction+MONEY_UNIT);
            return false;
        }
        return true;
    }

    public static final String MONEY_UNIT = " MGA";
}

File: MoneyUtils.js
class MoneyUtils {
    static isAmountValid(amount, balance = null, maxTransaction = null) {
        if (amount < 0) {
            console.log("Amount must be > 0 MGA");
            return false;
        }

        if (balance !== null && amount > balance) {
            console.log("Amount must be <= Balance:", balance, "MGA");
            return false;
        }

        if (maxTransaction !== null && amount > maxTransaction) {
            console.log("Amount must be <= Max Transaction:", maxTransaction, "MGA");
            return false;
        }

        return true;
    }
}

module.exports = { MoneyUtils };
